manifests/params.pp

```diff
@@ -9,6 +9,7 @@
   $install_options        = []
   $sources                = undef
   $puppet_master          = true
+  $proxy                  = ''
```

Comment:
I see that both undef and an empty param e.g [] or '' are used throughout the code base,  I like setting an empty string because it indicates what type of value is expected and when writing tests we can specifically check for an empty string instead of an undef value.  I'm easy either way though. . ..  .

Reply:
Empty hashes and arrays make more sense but I'd say not for strings.

Reply:
to quote our [review docs](https://voxpupuli.org/docs/#reviewing-a-module-pr):

> Are there new params with datatype Hash or Array? If possible, they should default to empty Hash/Array instead of undef. You can also enforce the datastructure like Array[String]

For simple datatypes like String, please default to undef.